### ISSUE TITLE
Can't use an undefined value as an ARRAY reference at cloc line 1839, <GEN32> line 12.

### ISSUE THREAD

**Issue by dustinkirkland** [AlDanial/cloc#87]
Running latest cloc script from git:

$ cloc --list-file=/tmp/cloc --sum-reports                                                                                                                                                                
Can't use an undefined value as an ARRAY reference at /home/kirkland/bin/cloc line 1839, <GEN32> line 12.

Note that I've run cloc against 26,535 separate source code trees, and generated 26,535 separate cloc reports.  Now, I want to sum all of them together.

**Comment by AlDanial** [comment]
I'm unable to reproduce the problem.  Here's what I tried, starting in the cloc directory after a git clone:
<pre>
  cloc --out aa.count tests/inputs/aa
  cloc --out dd.count tests/inputs/dd
  echo "aa.count" >  list.txt
  echo "dd.count" >> list.txt
  cloc --list-file list.txt --sum-reports
</pre>
The last line produces the expected output.  What happens when you run these lines?

The "line 12" part of the error message you cite is curious.  It makes me suspect the 12th line of your list file, /tmp/cloc, might have unexpected characters.

**Comment by dustinkirkland** [comment]
So I've trimmed my list file down, and it works for up to the first 31 lines, but when added the 32nd line, I get the same problem.  Here is my list-file: http://paste.ubuntu.com/15930579/

And you can find a very small (6KB) tarball of those count files at: http://people.canonical.com/~kirkland/count-files.tar.gz

Can you reproduce with these inputs?

**Comment by AlDanial** [comment]
Your file list and tar ball were perfect, thank you; I was able to reproduce and fix the problem rapidly.

There's a small element of forehead slapping on this one:  the 32nd file in the list, srv/changelogs/pool/multiverse/w/w3-recs/w3-recs_20110107-1/cloc, was generated with cloc v1.60.  The current git version is 1.67 (a development version; releases have even numbers).  Between 1.60 and 1.67, I corrected the case letters in JavaScript from 'Javascript' to 'JavaScript'.  Therefore when your git version 1.67 encountered the 12 line inside this 32nd file citing 'Javascript', cloc failed at a hash lookup.

Now the hash lookup itself should have triggered a 'next' statement to ignore this line and move on to the next one but it wasn't coded properly (at least not for perl 5.18 and higher).  I fixed the code to give a reasonable warning message about ignoring the language it couldn't understand as well as the file this was found it.

Ideally, <code>--sum-reports</code> would be run with the same version of cloc that created the original counts, but I can understand how you wouldn't want to rerun 26,000 cases each time a new release of cloc comes out.
